plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestingOracleServer.java

```diff
@@ -13,43 +13,78 @@
  */
 package io.trino.plugin.oracle;
 
+import io.airlift.log.Logger;
 import io.trino.plugin.jdbc.BaseJdbcConfig;
 import io.trino.plugin.jdbc.ConnectionFactory;
 import io.trino.plugin.jdbc.DriverConnectionFactory;
 import io.trino.plugin.jdbc.RetryingConnectionFactory;
 import io.trino.plugin.jdbc.credential.StaticCredentialProvider;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
 import oracle.jdbc.OracleDriver;
 import org.testcontainers.containers.OracleContainer;
 import org.testcontainers.utility.MountableFile;
 
 import java.io.Closeable;
-import java.io.IOException;
 import java.sql.Connection;
 import java.sql.SQLException;
 import java.sql.Statement;
+import java.time.temporal.ChronoUnit;
 
+import static io.airlift.testing.Closeables.closeAllSuppress;
 import static io.trino.testing.TestingConnectorSession.SESSION;
 import static java.lang.String.format;
 
 public class TestingOracleServer
-        extends OracleContainer
         implements Closeable
 {
+    private static final Logger log = Logger.get(TestingOracleServer.class);
+
+    private static final RetryPolicy<Object> CONTAINER_RETRY_POLICY = new RetryPolicy<>()
+            .withBackoff(1, 5, ChronoUnit.SECONDS)
+            .withMaxAttempts(5)
+            .handleIf(throwable -> throwable instanceof RetryableException)
+            .onRetry(event -> log.warn(
+                    "Container initialization failed on attempt %s, will retry. Exception: %s",
+                    event.getAttemptCount(),
+                    event.getLastFailure().getMessage()));
+
     private static final String TEST_TABLESPACE = "trino_test";
 
     public static final String TEST_USER = "trino_test";
     public static final String TEST_SCHEMA = TEST_USER; // schema and user is the same thing in Oracle
     public static final String TEST_PASS = "trino_test_password";
 
+    private final TestingOracleContainer container;
+
     public TestingOracleServer()
     {
-        super("wnameless/oracle-xe-11g-r2");
+        container = Failsafe.with(CONTAINER_RETRY_POLICY).get(this::createContainer);
+    }
+
+    private TestingOracleContainer createContainer()
+            throws RetryableException
+    {
+        TestingOracleContainer container = new TestingOracleContainer("wnameless/oracle-xe-11g-r2");
+        container.withCopyFileToContainer(MountableFile.forClasspathResource("init.sql"), "/docker-entrypoint-initdb.d/init.sql");
 
-        withCopyFileToContainer(MountableFile.forClasspathResource("init.sql"), "/docker-entrypoint-initdb.d/init.sql");
+        container.start();
+        try {
+            setUpDatabase(container);
+        }
+        catch (Exception e) {
+            container.stop();
+            closeAllSuppress(e);
```

Comment:
`closeAllSuppress(e)` is no-op

did you mean `closeAllSuppresse(e, container::stop)` ?